tests/unit_tests/kd_ema_test.py

```diff
@@ -0,0 +1,142 @@
+        kd_model.train(self.kd_train_params)
+        reloaded_ema_model = kd_model.ema_model.ema
+        reloaded_net = kd_model.net
+
+        # trained ema == loaded ema (Should always be true as long as "ema=True" in train_params)
+        assert check_models_have_same_weights(ema_model, reloaded_ema_model)
+
+        # loaded net != trained net (since load_ema_as_net = True)
+        assert not check_models_have_same_weights(reloaded_net, net)
+
+        # loaded net == trained ema (since load_ema_as_net = True)
+        assert check_models_have_same_weights(reloaded_net, ema_model)
+
+        # loaded student ema == loaded student net (since load_ema_as_net = True)
+        assert check_models_have_same_weights(reloaded_ema_model.module.student, reloaded_net.module.student)
+
+        # loaded teacher ema == loaded teacher net (teacher always loads ema)
+        assert check_models_have_same_weights(reloaded_ema_model.module.teacher, reloaded_net.module.teacher)
+
+
+
+    def test_kd_ckpt_reload_net(self):
+        """Check that the KD model load correctly from checkpoint when "load_ema_as_net=False"."""
+
+        # Create a KD model and train it
+        kd_model = KDModel("test_kd_ema_ckpt_reload", device='cpu')
+        kd_model.connect_dataset_interface(self.dataset)
+        kd_model.build_model(student_architecture='resnet18',
+                             teacher_architecture='resnet50',
+                             student_arch_params={'num_classes': 1000},
+                             teacher_arch_params={'num_classes': 1000},
+                             checkpoint_params={'teacher_pretrained_weights': "imagenet"},
+                             run_teacher_on_eval=True, )
+        kd_model.train(self.kd_train_params)
+        ema_model = kd_model.ema_model.ema
+        net = kd_model.net
+
+        # Load the trained KD model
+        kd_model = KDModel("test_kd_ema_ckpt_reload", device='cpu')
+        kd_model.connect_dataset_interface(self.dataset)
+        kd_model.build_model(student_architecture='resnet18',
+                             teacher_architecture='resnet50',
+                             student_arch_params={'num_classes': 1000},
+                             teacher_arch_params={'num_classes': 1000},
+                             checkpoint_params={"load_checkpoint": True, "load_ema_as_net": False},
+                             run_teacher_on_eval=True, )
+
+        # TRAIN FOR 0 EPOCHS JUST TO SEE THAT WHEN CONTINUING TRAINING EMA MODEL HAS BEEN SAVED CORRECTLY
+        kd_model.train(self.kd_train_params)
+        reloaded_ema_model = kd_model.ema_model.ema
+        reloaded_net = kd_model.net
+
+        # trained ema == loaded ema (Should always be true as long as "ema=True" in train_params)
+        assert check_models_have_same_weights(ema_model, reloaded_ema_model)
+
+        # loaded net == trained net (since load_ema_as_net = False)
+        assert check_models_have_same_weights(reloaded_net, net)
```

Comment:
use self.assert as done in other tests please.

Reply:
done in b161c58e646436229b624f587e3052e36382221e